src/display/network_utils.js

```diff
@@ -36,6 +36,15 @@ function createHeaders(isHttp, httpHeaders) {
   return headers;
 }
 
+function getResponseOrigin(url) {
+  try {
+    return new URL(url).origin;
+  } catch {
+    // `new URL()` will throw on incorrect data.
+  }
+  return null;
```

Comment:
```suggestion
  // Notably, null is distinct from "null" string (e.g. from file:-URLs).
  return null;
```

Added extra comment to emphasize the difference. The original issue primarily mattered for http(s):-URLs because these are the only ones supporting range requests, so the exact result for non-http(s) URLs is not super important.